src/Dofs/ConstraintHandler.jl

```diff
@@ -59,6 +59,53 @@ struct ConstraintHandler{DH<:AbstractDofHandler,T}
     closed::ScalarWrapper{Bool}
 end
 
+"""
+    RHSData
+
+Stores the constrained columns and mean of the diagonal of stiffness matrix `A`.
+"""
+struct RHSData{T}
+    m::T
+    constrained_columns::SparseMatrixCSC{T, Int}
+end
+
+"""
+    get_rhs_data(ch::ConstraintHandler, A::SparseMatrixCSC) -> RHSData
+
+Returns the needed RHSData for apply_rhs!
+"""
+function get_rhs_data(ch::ConstraintHandler, A::SparseMatrixCSC)
+    m = meandiag(A)
+    constrained_columns = A[:, ch.prescribed_dofs]
+    return RHSData(m, constrained_columns)
+end
+
+"""
+    apply_rhs!(data::RHSData, f::AbstractVector, ch::ConstraintHandler, applyzero::Bool=false)
+
+Applies the boundary condition to the right-hand-side vector without modifying the stiffness matrix
+"""
+function apply_rhs!(data::RHSData, f::AbstractVector, ch::ConstraintHandler, applyzero::Bool=false)	
+    K = data.constrained_columns
+    @assert length(f) == 0 || length(f) == size(K, 1)
+    @boundscheck length(f) == 0 || checkbounds(f, ch.prescribed_dofs)
+
+	m = data.m
+    @inbounds for i in 1:length(ch.values)
+        d = ch.prescribed_dofs[i]
+        v = ch.values[i]
+        if !applyzero && v != 0
+            for j in nzrange(K, d)
```

Comment:
@I'm actually uncertain if this line should work for arbitrary cases

Reply:
I think this indeed is what causes your error. `nzrange(K, col)` returns the indices for column `col`. `ch.prescribed_dofs` contains the global prescribed dofs, but `K` only contains the prescribed columns and not the full matrix. So with `d` we can index into the complete `K`, but not into the version that is stored in `RHSDATA`. As you suggest in your bugfix that needs to be indexed with `i`. 

The reason that this did not show up in your test is that all prescribed values there are 0 - thus this branch is never reached.

Reply:
Thanks so much for your time :relaxed:  ! And good catch to wait for me to write the example, ha!